src/jit/codegenarmarch.cpp

```diff
@@ -1207,7 +1207,34 @@ void CodeGen::genMultiRegCallStoreToLocal(GenTreePtr treeNode)
     {
         // Right now the only enregistrable multi-reg return types supported are SIMD types.
         assert(varTypeIsSIMD(treeNode));
-        NYI("GT_STORE_LCL_VAR of a SIMD enregisterable struct");
+
+        regNumber dst = treeNode->gtRegNum;
+
+        for (unsigned i = 0; i < regCount; ++i)
+        {
+            var_types type = pRetTypeDesc->GetReturnRegType(i);
+            regNumber reg  = call->GetRegNumByIdx(i);
+            if (op1->IsCopyOrReload())
+            {
+                // GT_COPY/GT_RELOAD will have valid reg for those positions
+                // that need to be copied or reloaded.
+                regNumber reloadReg = op1->AsCopyOrReload()->GetRegNumByIdx(i);
+                if (reloadReg != REG_NA)
+                {
+                    reg = reloadReg;
+                }
+            }
+
+            assert(reg != REG_NA);
+            if (varTypeIsFloating(type))
+            {
+                getEmitter()->emitIns_R_R_I_I(INS_mov, emitTypeSize(type), dst, reg, i, 0);
```

Comment:
Adding this comment makes this instruction more understandable:

```
 // mov vector[index] to vector[index2]
 // mov to 'dst[0]' from 'reg[i]'
```

Reply:
I'm a bit confused about how this works. Could you clarify? I *think* that what's needed here is to move from `reg` to the appropriate part of `dst`, and I'm not sure how this is being accomplished.

Reply:
Sure.  @briansull comment is incorrect, so clarification is clearly needed.